### PR TITLE
Updating pytorch

### PR DESCRIPTION
**Pull request by ebetica**
Quick summary of the things I noticed:

- all/any return Scalars instead of Bool
- Many of the reduction operations used to return 1-dim 1-elem tensors, but now they return 0 dim Scalar Tensors, so many of the cat ops must be replaced with stacks
- Possibly more stuff that's gone unnoticed :<

### CODE REVIEW

File: tests/test.cpp [hunk 1]
```diff
@@ -1,8 +1,12 @@
+#include <ATen/Config.h>
```
Comment (jgehring) [inline tests/test.cpp]: Not needed?

**Comment by ebetica** [comment]
Updated description with summary, will land with CUDNN LSTM as a stack.

**Comment by jgehring** [comment]
> Possibly more stuff that's gone unnoticed :<

We'll hopefully find out with our unit tests...

File: tests/test.cpp [hunk 2]
```diff
@@ -481,7 +483,7 @@ std::map<std::string, std::function<void()>> construct_tests() {
 
    EXPECT(y.defined());
    EXPECT(x.sizes().vec() == y.sizes().vec());
-   EXPECT(x.eq(y).all());
+   EXPECT(x.eq(y).all().toCInt());
```
Comment (eugene-kharitonov) [inline tests/test.cpp]: I wonder if
bool Tensor::allclose()
is not more readable/fitting in those comparisons?

Comment (ebetica) [reply tests/test.cpp]: Yes, good find! This didn't exist when I last updated PyTorch.

**Pull request merged**